docs/codestyle.md

```diff
@@ -0,0 +1,58 @@
+# WikiTree code style
+
+## Code style
+
+we aim for consistency, generality, readability and reduction of git diffs. Similar language constructs are formatted with similar rules. Style configuration options are deliberately limited and rarely added. Previous formatting is taken into account as little as possible.
+
+## Basic rules
+
+| Rule | Value | Note |
+| ---  | :---: | --- |
+| Line length | 120 characters |  |
+| Indentation | 4 spaces |  |
+| Semicolons at the end of the line| yes |  |
+| Quote type | Double quotes | e.g. `myObj["propA"]` |
+| Trailing comma | optional | *If you want to add a new property, you can add a new line without modifying the previously last line if that line already uses a trailing comma. This makes version-control diffs cleaner and editing code might be less troublesome.* [[source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas)] |
+| Bracket spacing | yes | e.g. `function xyz(arg1){ console.log(arg1) }` |
+| Bracket line | yes | Put the > of a multi-line HTML (HTML, JSX, Vue, Angular) element at the end of the last line instead of being alone on the next line (does not apply to self closing elements). |
+| Arrow Function Parentheses | yes | Arrow functions can omit parentheses when they have exactly one parameter. In all other cases the parameter(s) must be wrapped in parentheses. This rule enforces the consistent use of parentheses in arrow functions. [[source](https://eslint.org/docs/latest/rules/arrow-parens)] |
+| Single Attribute Per Line | no | Enforce single attribute per line in HTML |
+
+### Formatters
+
+| Formatter | configuration file | Homepage | Editor integration |
+| --- | --- | --- | --- |
+| Prettier | [`.prettierrc`](../.prettierrc) | [https://prettier.io/](https://prettier.io/) | [VSCode](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode), [WebStorm](https://plugins.jetbrains.com/plugin/10456-prettier), [Sublime Text](https://packagecontrol.io/packages/JsPrettier) and lot of others... check the [official website](https://prettier.io/docs/en/editors.html)|
+
+## Naming conventions
+
+| Object | Convention |
+| --- | --- |
+| Variables |  `camel case` with lowercase first letter, e.g. `familyName`  |
+| Booleans | We should use `is` or `has` as prefixes, e.g. `isDead`, `hasWife` |
+| Functions | Same as with variables + should use descriptive nouns and verbs as prefixes, e.g. `getName()`, `generateReport` |
+| Constants | Should be written in `upper snake case`, e.g. `DEFAULT_COUNT_OF_GENERATIONS` |
+| Classes | Should be written in `Pascal case`, e.g. `WikiTreePerson` |
+| Methods | Same as functions |
+| Files | File names must be all lowercase and may include underscores (_) or dashes (-), but no additional punctuation. Follow the convention that your project uses. [[source](https://google.github.io/styleguide/jsguide.html#file-name)] - most Unix-based servers are case sensitives, but this doesn't apply for windows servers.|
```

Comment:
Filenames - this needs discussion and consensus...

Reply:
I'm torn on the filenames. I get the Google style guide suggestion. But I sort of like filenames that match their contents. For example, if there's a JS file that just contains a particular class, I like naming it ClassName.js, matching the "obj = new ClassName" formatting of the class itself. And from my own habits, I find if I use snake_case to make a long filename readable, I end up using that for variables sometimes too, rather than being consistently camelCase.

But obviously filenames will end up with a mixture of formats if we use both PascalCase and camelCase for the names, depending on whether it's a Class file or not. And I suppose all-lower-case filenames may be helpful for people moving from one OS to another (I've run into issues in OSX with TLA due to filename case expectations).

I'm curious if anyone else has any opinions on the matter.